### PR TITLE
WIP: Updated .travis.yml To Use Native 18.04 Support

### PR DESCRIPTION
**Pull request by garethellis0**
### Description

- removed docker dependency
- changed travis ci to use ubuntu 18.04 natively
- removed `travis_ci.sh` and created `clang_formatting/check_formatting.sh` to contain the functionality for checking formatting
- moved the rest of the `travis_ci.sh` functionality, while simultaneously making `.travis.yml` way shorter and more readable

### Testing Done
CI passes, checked that failures in the formatting script actually fail CI

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,55 +6,24 @@
...
-# The variables here are uniquely set at the start of each different test
-matrix:
-  include:
-  - name: "ROS Melodic / Ubuntu 18.04 - Build and Test"
-    env: RUN_BUILD="true"
-         RUN_TESTS="true"
-         RUN_COVERAGE="true"
-  - name: "ROS Melodic / Ubuntu 18.04 - Check Formatting"
-    env: RUN_FORMATTING_CHECKS="true"
+script:
```
Comment (MathewMacDougall) [inline .travis.yml]: without the matrix are we no longer running tests / formatting? or am I misunderstanding?

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (garethellis0) [reply .travis.yml]: _Holy, didn't even request a review yet_

Just playing around with stuff now, but yeah, will need to add coverage.

Comment (garethellis0) [reply .travis.yml]: Oh, I see github requested a review _for_ me. I'll let you know when this needs review.

**Comment by garethellis0** [comment]
I have no idea how coverage dropped. I can only guess our old CI process somehow hid poor coverage.

File: .travis.yml (hunk 1 above)
Comment (MathewMacDougall) [reply .travis.yml]: Sorry lol, if it shows up in my inbox I review it

Comment (garethellis0) [reply .travis.yml]: Haha same

**Comment by garethellis0** [comment]
@MathewMacDougall @akhilveeraghanta - this is good to review now

**Review by MathewMacDougall — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]
Looks Good :tm:

**Pull request merged**